internal/services/rewards.go

```diff
@@ -292,11 +296,40 @@ func (t *BaselineClient) assignPoints() error {
 			}
 		}
 
+		// TODO(ae): need to add signature here still, must update devicees-api grpc response to do so
+		earningVehicle, ok := attestationData[*ud.Vin]
+		if ok {
+			credExp, ok := earningVehicle[CredentialExpiresAt]
+			if ok {
+				if credExp.(time.Time).Before(ud.LatestVinCredential.Expiration.AsTime()) {
+					attestationData[*ud.Vin] = map[string]interface{}{
+						CredentialExpiresAt: ud.LatestVinCredential.Expiration.AsTime(),
+						Values:              []interface{}{*ud.TokenId, ud.DeviceDefinitionId, ud.LatestVinCredential.Id},
+					}
+				}
+
+			}
+		}
+		attestationData[*ud.Vin] = map[string]interface{}{
+			CredentialExpiresAt: ud.LatestVinCredential.Expiration.AsTime(),
+			Values:              []interface{}{*ud.TokenId, ud.DeviceDefinitionId, ud.LatestVinCredential.Id},
+		}
```

Comment:
Could these all be structs? I don't see the advantage of passing all these maps and arrays around.

Reply:
fair point, updated this to be a struct which also made the func signature of generateMerkleTree less odd